src/main/scala/io/citrine/lolo/trees/Nodes.scala

```diff
@@ -34,7 +34,7 @@ abstract class TrainingNode[T <: AnyVal, S](
   def getFeatureImportance(): mutable.ArraySeq[Double]
 }
 
-trait ModelNode[T <: PredictionResult[Any]] extends Serializable {
+trait ModelNode[+T <: PredictionResult[Any]] extends Serializable {
```

Comment:
Making the type covariant indicates that, for example, `InternalModelNode[PredictionResult[Double]]` is a subtype of `ModelNode[PredictionResult[Any]]`, and removes the need for a typecast.